tests/templates/test_state_preparations/test_basis_state_prep.py

```diff
@@ -0,0 +1,165 @@
+        ([1, 0], [1, 4], [1]),
+        ([1, 1], [0, 2], [0, 2]),
+        ([1, 0], [4, 5], [4]),
+        ([0, 0, 1, 0], [1, 2, 3, 4], [3]),
+        ([1, 1, 1, 0], [1, 2, 6, 8], [1, 2, 6]),
+        ([1, 0, 1, 1], [1, 2, 6, 8], [1, 6, 8]),
+    ])
+    # fmt: on
+    def test_correct_pl_gates(self, basis_state, wires, target_wires):
+        """Tests queue for simple cases."""
+
+        op = qml.templates.BasisStatePreparation(basis_state, wires)
+        queue = op.expand().operations
+
+        for id, gate in enumerate(queue):
+            assert gate.name == "PauliX"
+            assert gate.wires.tolist() == [target_wires[id]]
+
+    # fmt: off
+    @pytest.mark.parametrize("basis_state,wires,target_state", [
+        ([0], [0], [0, 0, 0]),
+        ([0], [1], [0, 0, 0]),
+        ([1], [0], [1, 0, 0]),
+        ([1], [1], [0, 1, 0]),
+        ([0, 1], [0, 1], [0, 1, 0]),
+        ([1, 1], [0, 2], [1, 0, 1]),
+        ([1, 1], [1, 2], [0, 1, 1]),
+        ([1, 0], [0, 2], [1, 0, 0]),
+        ([1, 1, 0], [0, 1, 2], [1, 1, 0]),
+        ([1, 0, 1], [0, 1, 2], [1, 0, 1]),
+    ])
+    # fmt: on
+    def test_state_preparation(self, tol, qubit_device_3_wires, basis_state, wires, target_state):
+        """Tests that the template produces the correct expectation values."""
+
+        @qml.qnode(qubit_device_3_wires)
+        def circuit():
+            qml.templates.BasisStatePreparation(basis_state, wires)
+
+            # Pauli Z gates identify the basis state
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1)), qml.expval(qml.PauliZ(2))
+
+        # Convert from Pauli Z eigenvalues to basis state
+        output_state = [0 if x == 1.0 else 1 for x in circuit()]
+
+        assert np.allclose(output_state, target_state, atol=tol, rtol=0)
+
+    def test_custom_wire_labels(self, tol):
+        """Test that template can deal with non-numeric, nonconsecutive wire labels."""
+        basis_state = [0, 1, 0]
+
+        dev = qml.device("default.qubit", wires=3)
+        dev2 = qml.device("default.qubit", wires=["z", "a", "k"])
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.templates.BasisStatePreparation(basis_state, wires=range(3))
+            return qml.expval(qml.Identity(0))
+
+        @qml.qnode(dev2)
+        def circuit2():
+            qml.templates.BasisStatePreparation(basis_state, wires=["z", "a", "k"])
+            return qml.expval(qml.Identity("z"))
+
+        circuit()
+        circuit2()
+
+        assert np.allclose(dev.state, dev2.state, atol=tol, rtol=0)
+
+
+class TestInputs:
+    """Test inputs and pre-processing."""
+
+    def test_exception_wrong_dim(self):
+        """Verifies that exception is raised if the
+        number of dimensions of features is incorrect."""
+        dev = qml.device("default.qubit", wires=3)
+
+        @qml.qnode(dev)
+        def circuit(weights):
+            qml.templates.ArbitraryStatePreparation(weights, wires=range(3))
```

Comment:
I believe this test is in the wrong file.

Reply:
Ah, good find!